_studio/mfx_lib/shared/src/mfx_h264_encode_vaapi.cpp

```diff
@@ -309,6 +313,55 @@ static mfxStatus SetMaxFrameSize(
     return MFX_ERR_NONE;
 }
 
+static mfxStatus SetMultiPassFrameSize(
+    DdiTask const & task,
+    mfxU32       fieldId,
+    VADisplay    vaDisplay,
+    VAContextID  vaContextEncode,
+    VABufferID & frameSizeBuf_id)
+{
+    VAEncMiscParameterBuffer             *misc_param;
+    VAEncMiscParameterBufferMultiPassFrameSize *p_multiPassFrameSize;
+
+    mfxStatus mfxSts = CheckAndDestroyVAbuffer(vaDisplay, frameSizeBuf_id);
+    MFX_CHECK_STS(mfxSts);
+
+    VAStatus vaSts = vaCreateBuffer(vaDisplay,
+                   vaContextEncode,
+                   VAEncMiscParameterBufferType,
+                   sizeof(VAEncMiscParameterBuffer) + sizeof(VAEncMiscParameterBufferMultiPassFrameSize),
+                   1,
+                   NULL,
+                   &frameSizeBuf_id);
+    MFX_CHECK_WITH_ASSERT(VA_STATUS_SUCCESS == vaSts, MFX_ERR_DEVICE_FAILED);
+
+    vaSts = vaMapBuffer(vaDisplay, frameSizeBuf_id, (void **)&misc_param);
+    MFX_CHECK_WITH_ASSERT(VA_STATUS_SUCCESS == vaSts, MFX_ERR_DEVICE_FAILED);
+
+    misc_param->type = VAEncMiscParameterTypeMultiPassFrameSize;
+    p_multiPassFrameSize = (VAEncMiscParameterBufferMultiPassFrameSize *)misc_param->data;
+
+    if (task.m_type[fieldId] & MFX_FRAMETYPE_I)
+        p_multiPassFrameSize->max_frame_size = task.m_repackMaxIFrameSize;
+    else if (task.m_type[fieldId] & MFX_FRAMETYPE_P)
+        p_multiPassFrameSize->max_frame_size = task.m_repackMaxPFrameSize;
+    else if (task.m_type[fieldId] & MFX_FRAMETYPE_B)
+        p_multiPassFrameSize->max_frame_size = task.m_repackMaxBFrameSize;
+    else // bad coding type
+        return MFX_ERR_DEVICE_FAILED;
+
+    p_multiPassFrameSize->num_passes = task.m_repackNumPasses;
+    p_multiPassFrameSize->delta_qp = (unsigned char *)task.m_repackDeltaQP;
+
+    {
+        MFX_AUTO_LTRACE(MFX_TRACE_LEVEL_EXTCALL, "vaUnmapBuffer");
```

Comment:
if you trace unmap, then add a trace to map, or remove both

Reply:
Will remove the trace here.